napari_animation/_tests/test_animation.py

```diff
@@ -139,3 +145,48 @@ def test_end_state_reached(image_animation):
     image_animation.capture_keyframe(steps=2)
     last_state = image_animation._frames[-1]
     assert last_state == image_animation.key_frames[-1].viewer_state
+
+
+@pytest.mark.parametrize("layer_class, data, ndim", layer_test_data)
+def test_attributes_for_all_layer_types(
+    make_napari_viewer, layer_class, data, ndim
+):
+    """Test that attributes are in the viewer_state for all napari layer types"""
+    viewer = make_napari_viewer()
+    add_layer_by_type(viewer, layer_class, data, visible=True)
+    layer_animation = Animation(viewer)
+    # get the state of the layer
+    layer_state = viewer.layers[0]._get_state()
+    # remove attributes that arn't captured
+    layer_state.pop("metadata")
+    layer_state.pop("ndim", None)
+    layer_state.pop("property_choices", None)
+    layer_state.pop("colormaps_dict", None)
+    layer_state.pop("data")
+    # these have `allow_mutation=False` so can't be set
+    if layer_class == Surface:
+        layer_state["normals"]["face"].pop("mode", None)
+        layer_state["normals"]["vertex"].pop("mode", None)
+
+    layer_animation.capture_keyframe()
+    # get the layer attributes captured to viewer_state
+    animation_state = layer_animation.key_frames[0].viewer_state.layers[
+        viewer.layers[0].name
+    ]
+
+    assert_layer_state_equal(animation_state, layer_state)
+
+
+@pytest.mark.parametrize("layer_class, data, ndim", layer_test_data)
+def test_animating_all_layer_types(
+    make_napari_viewer, layer_class, data, ndim
+):
+    """Test that all napari layer types can be animated"""
+    viewer = make_napari_viewer()
+    add_layer_by_type(viewer, layer_class, data, visible=True)
+    layer_animation = Animation(viewer)
+    layer_animation.capture_keyframe()
+    layer_animation.viewer.camera.zoom *= 2
+    layer_animation.capture_keyframe()
+    # advance the movie frame, simulating slider movement
+    layer_animation.set_movie_frame_index(1)
```

Comment:
Should this test assert something? like that an exception is not thrown? I mean if there is an exception then it's a fail--like on main with this test--so maybe this is fine?

Reply:
I think it's just checking that there's no exception, yes. You could add an assert later that the viewer did indeed change, but maybe that's tested elsewhere?